app/models/document_type.rb

```diff
@@ -3,9 +3,8 @@
 class DocumentType
   include InitializeWithHash
 
-  attr_reader :contents, :id, :label, :managed_elsewhere, :publishing_metadata,
-              :path_prefix, :tags, :guidance_govspeak, :description, :hint_govspeak, :images,
-              :topics, :check_path_conflict
+  attr_reader :contents, :id, :managed_elsewhere, :publishing_metadata, :label,
```

Comment:
Agreed it would be great to extract this. If we move it back into I18n, the problem is the dynamically generated document types lead to a `translation missing: document_types.abc123.description` type error. We'd have to make a choice:

1) convert `document_type.description` to `t("document_types.#{document_type.id}.description", default: "")` - downside being introducing a `default` that is only ever used by our tests; something we considered and rejected for `label`.
2) inject I18n key into our tests every time we generate a document type; something we considered and rejected for `label`.

I couldn't think of a way of doing it that we hadn't already dismissed in the case of `label`, so for consistency moved it back out 😞

Reply:
Ah, so I think I wasn't clear enough about what I meant by 'optional'. I meant doing [something like this](https://github.com/alphagov/content-publisher/pull/1659/files#diff-7ee41565bf7db9c2b08e740bd04bfe64R23) - guess that's option 3 😉.

The factory illustrates that it's optional, in that we've been creating document types with no description - the ternary would be equivalent to that.

Does that make more sense?

Reply:
Hmm that _is_ an option 3 - but is still using application code to cater for a situation we'll only encounter in our tests. For me it's equivalent to option 1.

So I guess we need to decide if it's better to have some redundant 'optional' logic in the view and a cleaner separation of 'description' into I18n, or a cleaner view but some icky config 🤔 I'm starting to think option 1 is the best option - what do you think?

Reply:
So you're right that the app code is catering for the tests. That's not a bad thing in itself - we do it all the time with class interfaces - as long as it doesn't surprise/confuse people. If we went with option 1, my only concern is that it's inconsistent with all other translations (i.e. something new), which would be more surprising than the ternary pattern we have elsewhere. In itself, seeing code to cope with an optional description wouldn't surprise me.

The problem with `label` is that it's not optional: we have to put something there, and all of the alternatives were surprising. Conversely, it's plausible that a document type may not have a description. Do agree we should avoid option 2!

I think the case is pretty strong, but I also think we could leave it as-is for now and look at it in a separate PR, so it doesn't block this one going in.